modular_skyrat/modules/morenarcotics/code/reagents/opium.dm

```diff
@@ -0,0 +1,184 @@
+	reagent_state = LIQUID
+	color = "#ffe669"
+	overdose_threshold = 20
+	ph = 6
+	taste_description = "flowers"
+	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
+	inverse_chem = /datum/reagent/drug/blacktar/liquid
+	addiction_types = list(/datum/addiction/opioids = 66)
+
+/datum/reagent/drug/heroin/on_mob_life(mob/living/carbon/M, seconds_per_tick, times_fired)
+	var/high_message = pick("You feel like nothing can stop you.", "You feel like God.")
+	var/strength_multiplier = creation_purity**2
+	if(SPT_PROB(2.5, seconds_per_tick))
+		to_chat(M, span_notice("[high_message]"))
+	M.add_mood_event("smacked out", /datum/mood_event/narcotic_heavy, name)
+	M.adjustBruteLoss(-0.5 * strength_multiplier * REM * seconds_per_tick, 0) //more powerful as a painkiller, possibly actually useful to medical now
+	M.adjustFireLoss(-0.5 * strength_multiplier * REM * seconds_per_tick, 0)
+	M.adjustToxLoss(0.1 * (1/strength_multiplier) * REM * seconds_per_tick, 0)
+	M.overlay_fullscreen("heroin_euphoria", /atom/movable/screen/fullscreen/color_vision/heroin_color)
+	..()
+
+/datum/reagent/drug/heroin/overdose_process(mob/living/M, seconds_per_tick, times_fired)
+	M.adjustOrganLoss(ORGAN_SLOT_BRAIN, 0.75 * REM * seconds_per_tick)
+	M.adjustToxLoss(1.5 * REM * seconds_per_tick, 0)
+	M.adjust_drowsiness(1 SECONDS * REM * normalise_creation_purity() * seconds_per_tick)
+	return TRUE
+
+/datum/reagent/drug/heroin/on_mob_metabolize(mob/living/metabolizer)
+	. = ..()
+	metabolizer.apply_status_effect(/datum/status_effect/grouped/screwy_hud/fake_healthy, type)
+
+/datum/reagent/drug/heroin/on_mob_end_metabolize(mob/living/metabolizer)
+	. = ..()
+	metabolizer.remove_status_effect(/datum/status_effect/grouped/screwy_hud/fake_healthy, type)
+	metabolizer.clear_fullscreen("heroin_euphoria")
+
+/datum/reagent/drug/heroin/powder_heroin
+	name = "Powder heroin"
+	description = "The powder form of heroin."
+
+/datum/reagent/drug/blacktar
+	name = "Black tar heroin"
+	description = "An impure, freebase form of heroin. Probably not a good idea to take this..."
+	reagent_state = LIQUID
+	color = "#242423"
+	overdose_threshold = 10 //more easy to overdose on
+	ph = 8
+	taste_description = "flowers"
+	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
+	metabolization_rate = 0.05
+	addiction_types = list(/datum/addiction/opioids = 66)
+
+/datum/reagent/drug/blacktar/on_mob_life(mob/living/carbon/M, seconds_per_tick, times_fired)
+	var/high_message = pick("You feel like tar.", "The blood in your veins feel like syrup.")
+	if(SPT_PROB(2.5, seconds_per_tick))
+		to_chat(M, span_notice("[high_message]"))
+	M.add_mood_event("smacked out", /datum/mood_event/narcotic_heavy, name)
+	M.set_drugginess(20 SECONDS * REM * seconds_per_tick)
+	M.adjustBruteLoss(-0.1 * REM * seconds_per_tick, 0)
+	M.adjustFireLoss(-0.1 * REM * seconds_per_tick, 0)
+	M.adjustToxLoss(0.5 * REM * seconds_per_tick, 0) //toxin damage
+	return ..()
```

Comment:
I'm not going to suggest it, just do it like the other ones, I'm tired of re-doing the same suggestions.
Also. No single-letter variables.